redfish-core/lib/log_services.hpp

```diff
@@ -4154,6 +4154,85 @@ inline void getSystemHardwareIsolationLogEntryById(
         objMapperIface, "GetObject", entryObjPath.str, hwIsolationEntryIfaces);
 }
 
+/**
+ * @brief API Used to deisolate the given HardwareIsolation entry.
+ *
+ * @param[in] req - The HardwareIsolation redfish request (unused now).
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryId - The entry id of HardwareIsolation entries to deisolate.
+ *
+ * @return The redfish response in the given buffer.
+ */
+inline void deleteSystemHardwareIsolationLogEntryById(
+    const crow::Request& /* req */,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& entryId)
+{
+    sdbusplus::message::object_path entryObjPath(
+        std::string(hwIsolationEntryRootDbusPath) + "/" + entryId);
+
+    // Make sure the given entry id is present in hardware isolation
+    // entries and get the DBus name of that entry
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, entryId, entryObjPath](const boost::system::error_code ec,
+                                           const GetObjectType& objType) {
+            if (ec || objType.empty())
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value()
+                                 << " : " << ec.message()
+                                 << "] when tried to get the HardwareIsolation "
+                                    "dbus name the given object path: "
+                                 << entryObjPath.str;
+                if (ec.value() == EBADR)
+                {
+                    messages::resourceNotFound(asyncResp->res, "Entry",
+                                               entryId);
+                }
+                else
+                {
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
+
+            if (objType.size() > 1)
+            {
+                BMCWEB_LOG_ERROR << "More than one dbus service implemented "
+                                    "the HardwareIsolation service";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (objType[0].first.empty())
```

Comment:
duplicate check?
Its in many places. Please check and remove

Reply:
Yes, we want to check here